accepted/2020/platform-checks/platform-checks.md

```diff
@@ -0,0 +1,511 @@
+    {
+        public AddedInOSPlatformVersionAttribute(string osPlatform,
+                                                 int major,
+                                                 int minor,
+                                                 int build,
+                                                 int revision);
+    }
+
+    [AttributeUsage(AttributeTargets.Assembly |
+                    AttributeTargets.Class |
+                    AttributeTargets.Constructor |
+                    AttributeTargets.Event |
+                    AttributeTargets.Method |
+                    AttributeTargets.Module |
+                    AttributeTargets.Property |
+                    AttributeTargets.Struct,
+                    AllowMultiple=false, Inherited=true)]
+    public sealed class RemovedInOSPlatformVersionAttribute  : OSPlatformVersionAttribute
+    {
+        public RemovedInOSPlatformVersionAttribute(string osPlatform,
+                                                   int major,
+                                                   int minor,
+                                                   int build,
+                                                   int revision);
+    }
+
+    [AttributeUsage(AttributeTargets.Assembly |
+                    AttributeTargets.Class |
+                    AttributeTargets.Constructor |
+                    AttributeTargets.Event |
+                    AttributeTargets.Method |
+                    AttributeTargets.Module |
+                    AttributeTargets.Property |
+                    AttributeTargets.Struct,
+                    AllowMultiple=false, Inherited=true)]
+    public sealed class ObsoletedInOSPlatformVersionAttribute  : OSPlatformVersionAttribute
+    {
+        public ObsoletedInOSPlatformVersionAttribute(string osPlatform,
+                                              int major,
+                                              int minor,
+                                              int build,
+                                              int revision);
+        public string Url { get; set; }
+    }
+}
+```
+
+### Platform context
+
+To determine the platform context of the call site the analyzer must consider
+the following two configurations:
+
+1. **Call-site**. Application of `AddedInOSPlatformVersionAttribute` to the containing
+   member, type, module, or assembly.
+
+2. **Project** The project TFMs and `TargetPlatformMinVersion`. If the project
+   has an platform-specific TFM (such as `net5.0-ios13.0`) we consider it to be
+   on the specified platform. If `TargetPlatformMinVersion` is set, we assume
+   that version, otherwise we assume the version as specified by the TFM.
+
+If (2) is platform neutral (e.g. `net5.0`), the platform context is entirely
+determined by (1).
+
+If (2) is platform-specific it is combined with (1):
+
+* If the platforms between (1) and (2) are disjoint (such as iOS and Windows),
+  the information from (1) is discarded.
```

Comment:
I am not following this, if {1} API annotated with iOS specific and project {2} TFM is  `net5.0-windows` then shouldn't we warn that the API is for {1} iOS? Why the information from {1} is discarded?